easybuild/tools/build_log.py

```diff
@@ -129,13 +129,16 @@ def deprecated(self, msg, ver, *args, **kwargs):
                     else: version to check against max_ver to determine warning vs exception
         :param max_ver: version threshold for warning vs exception (compared to 'ver')
         """
-        max_ver = kwargs.get('max_ver', None)
+        max_ver = kwargs.pop('max_ver', None)
+        if max_ver is None and args:
+            max_ver = args[0]
+            args = args[1:]
```

Comment:
you have no way to know that max_ver will be args[0], so please don't do this, if you expect it, 
make the function definition 
`def deprecated(self, msg, ver, max_ver=None,*args, **kwargs):`
`